### PR TITLE
Adds Scalyr Logging Endpoint Support

### PR DESCRIPTION
**Pull request by mccurdyc**
## Proposed Changes

* Adds Scalyr logging endpoint support.

## Relevant Link(s) / Documentation

* [API Docs](https://developer.fastly.com/reference/api/logging/scalyr/)

## Validation

`export FASTLY_API_KEY=foo; export TF_ACC=1; make testacc`

cc @phamann @ezkl

### CODE REVIEW

**Comment by mccurdyc** [comment]
Force-pushed two squashed commits that make the Token field required.

File: fastly/block_fastly_service_v1_logging_scalyr.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+			// Required fields
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique name of the Scalyr logging endpoint.",
+			},
+
+			"token": {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_scalyr.go:26]: The docs don't say required, but it's required in the [CLI](https://github.com/fastly/cli/blob/0264e0b19e942ad082a25506567a82aa96fe8116/pkg/logging/scalyr/create.go#L43) and confirmed that this is required by the backend.

Also noted that the docs need to be updated.

**Comment by mccurdyc** [comment]
Force-pushed a squashed commit that fixed a grammar error in the docs and renames variables using 'h'.

File: fastly/block_fastly_service_v1_logging_scalyr.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		return err
+	}
+
+	return nil
+}
+
+func flattenScalyr(scalyrList []*gofastly.Scalyr) []map[string]interface{} {
+	var flattened []map[string]interface{}
+	for _, s := range scalyrList {
+		// Convert HTTP logging to a map for saving to state.
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_scalyr.go]: TODO @mccurdyc fix copy-paste error.

**Comment by mccurdyc** [comment]
TODO: add logging_scalyr to this block - https://github.com/terraform-providers/terraform-provider-fastly/pull/252/files#diff-e11f97f0c242e77100f2f794b0ac08abR1578

File: fastly/block_fastly_service_v1_logging_scalyr.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+	if err := d.Set("logging_scalyr", scalyrLogList); err != nil {
+		log.Printf("[WARN] Error setting Scalyr logging endpoints for (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func createScalyr(conn *gofastly.Client, i *gofastly.CreateScalyrInput) error {
+	_, err := conn.CreateScalyr(i)
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_scalyr.go:134]: This should be simplified to be `return err`

**Comment by mccurdyc** [comment]
Force-pushed a squashed commit addressing self-review comments.

**Comment by mccurdyc** [comment]
Force-pushed a squashed commit addressing more self-review comments.

File: fastly/resource_fastly_service_v1.go [hunk 4]
```diff
@@ -3526,6 +3535,11 @@ func resourceServiceV1Read(d *schema.ResourceData, meta interface{}) error {
 			return err
 		}
 
+		// Refresh Scalyr
+		if err := readScalyr(conn, d, s); err != nil {
+			return err
+		}
```
Comment (phamann) [inline fastly/resource_fastly_service_v1.go]: I'm interested, your other PR's for new endpoints haven't included this read. If correct we should fix in the others.

**Review by phamann — APPROVED** [review]
👍 LGTM - other than non-blocking comment regarding read.

**Comment by mccurdyc** [comment]
Force-pushed squashing into a single commit.

**Comment by mccurdyc** [comment]
Force-pushed rebasing master.

File: fastly/resource_fastly_service_v1.go (hunk 4 above)
Comment (mccurdyc) [reply fastly/resource_fastly_service_v1.go]: Absolutely! I think they should now. I will confirm.

**Pull request merged**